src/test/java/com/purbon/kafka/topology/integration/SaslPlaintextKafkaContainer.java

```diff
@@ -0,0 +1,142 @@
+package com.purbon.kafka.topology.integration;
+
+import com.github.dockerjava.api.command.ExecCreateCmdResponse;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.Admin;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.DockerImageName;
+
+/** @author <a href="mailto:shh@thathost.com">Sverre H. Huseby</a> */
+final class SaslPlaintextKafkaContainer extends GenericContainer<SaslPlaintextKafkaContainer> {
+
+  private static final DockerImageName DEFAULT_IMAGE =
+      DockerImageName.parse("confluentinc/cp-kafka").withTag("6.0.0");
+  public static final int KAFKA_PORT = 9092;
+  public static final int ZOOKEEPER_PORT = 2181;
+  private static final int PORT_NOT_ASSIGNED = -1;
+  /* Note difference with 0.0.0.0 and localhost: The former will be replaced by the container IP. */
+  private static final String LISTENERS =
+      "SASL_PLAINTEXT://0.0.0.0:9092,INTERNAL://127.0.0.1:29092";
+  private int port = PORT_NOT_ASSIGNED;
+  private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
+
+  public SaslPlaintextKafkaContainer() {
+    this(DEFAULT_IMAGE);
+  }
+
+  public SaslPlaintextKafkaContainer(final DockerImageName dockerImageName) {
+    super(dockerImageName);
+    withExposedPorts(KAFKA_PORT, ZOOKEEPER_PORT);
+    withEnv("KAFKA_BROKER_ID", "1");
+    withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1");
+    withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", "1");
+    withEnv("KAFKA_LOG_FLUSH_INTERVAL_MESSAGES", Long.MAX_VALUE + "");
+    withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0");
+    withEnv("KAFKA_LISTENERS", LISTENERS);
+    withEnv(
+        "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP",
+        "SASL_PLAINTEXT:SASL_PLAINTEXT,INTERNAL:SASL_PLAINTEXT");
+    withEnv("KAFKA_INTER_BROKER_LISTENER_NAME", "INTERNAL");
+    withEnv("KAFKA_SASL_MECHANISM_INTER_BROKER_PROTOCOL", "PLAIN");
+    withEnv("KAFKA_SASL_ENABLED_MECHANISMS", "PLAIN");
+    withEnv("KAFKA_AUTHORIZER_CLASS_NAME", "kafka.security.authorizer.AclAuthorizer");
+    withEnv("KAFKA_SUPER_USERS", "User:kafka");
```

Comment:
What do you think about adding the super user as parameter? we can keep kafka as default, but thinking out loud if it makes sense to have this extended , what do you think?

Reply:
I guess that would have to be reflected in the JAAS configs as well, including a matching password. Since this class was made for KTB, I just replicated the test user setup from your docker-compose files. If you insist, I will consider .withSuperUser(username, pasword) and a repeatable .withUser(username, password).

Reply:
I am fine with keeping Kafka as a super user now, I was just putting an idea, might be something to extend in the future if necessary. But don't expend much time on it :-) I hope it makes sense.